service/controller/resource/pvc/current.go

```diff
@@ -19,25 +18,15 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 
 	r.logger.Debugf(ctx, "looking for PVCs in the Kubernetes API")
 
-	var PVCs []*corev1.PersistentVolumeClaim
-
 	namespace := key.ClusterNamespace(customObject)
-	pvcNames := key.PVCNames(customObject)
-
-	for _, name := range pvcNames {
-		manifest, err := r.k8sClient.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, name, metav1.GetOptions{})
-		if apierrors.IsNotFound(err) {
-			r.logger.Debugf(ctx, "did not find a PVC in the Kubernetes API")
-			// fall through
-		} else if err != nil {
-			return nil, microerror.Mask(err)
-		} else {
-			r.logger.Debugf(ctx, "found a PVC in the Kubernetes API")
-			PVCs = append(PVCs, manifest)
-		}
+	pvcs, err := r.k8sClient.CoreV1().PersistentVolumeClaims(namespace).List(ctx, metav1.ListOptions{
```

Comment:
Main fix is here. List all cluster PVCs instead of calculating names and getting each by name. This way we pick up local volume PVCs as needing to be deleted so we can unbind them from the PV.